pkg/kubectl/cmd/auth/reconcile.go

```diff
@@ -0,0 +1,220 @@
+			return cmdutil.PrintResourceInfoForCommand(cmd, info, f, o.Out)
+		}
+		cmdutil.PrintSuccess(mapper, shortOutput, o.Out, info.Mapping.Resource, info.Name, dryRun, "reconciled")
+		return nil
+	}
+
+	return nil
+}
+
+func (o *ReconcileOptions) Validate() error {
+	return nil
+}
+
+func (o *ReconcileOptions) RunReconcile() error {
+	r := o.ResourceBuilder.Do()
+	err := r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// shallowInfoCopy this is used to later twiddle the Object for printing
+		// we really need more straightforward printing options
+		shallowInfoCopy := *info
+
+		switch t := info.Object.(type) {
+		case *rbac.Role:
+			reconcileOptions := reconciliation.ReconcileRoleOptions{
+				Confirm:                true,
+				RemoveExtraPermissions: false,
+				Role: reconciliation.RoleRuleOwner{Role: t},
+				Client: reconciliation.RoleModifier{
+					NamespaceClient: o.CoreClient.Namespaces(),
+					Client:          o.RBACClient,
+				},
+			}
+			result, err := reconcileOptions.Run()
+			if err != nil {
+				return err
+			}
+			shallowInfoCopy.Object = result.Role.GetObject()
+			o.Print(&shallowInfoCopy)
+			return nil
+
+		case *rbac.ClusterRole:
+			reconcileOptions := reconciliation.ReconcileRoleOptions{
+				Confirm:                true,
+				RemoveExtraPermissions: false,
+				Role: reconciliation.ClusterRoleRuleOwner{ClusterRole: t},
+				Client: reconciliation.ClusterRoleModifier{
+					Client: o.RBACClient.ClusterRoles(),
+				},
+			}
+			result, err := reconcileOptions.Run()
+			if err != nil {
+				return err
+			}
+			shallowInfoCopy.Object = result.Role.GetObject()
+			o.Print(&shallowInfoCopy)
+			return nil
+
+		case *rbac.RoleBinding:
+			reconcileOptions := reconciliation.ReconcileRoleBindingOptions{
+				Confirm:             true,
+				RemoveExtraSubjects: false,
+				RoleBinding:         reconciliation.RoleBindingAdapter{RoleBinding: t},
+				Client: reconciliation.RoleBindingClientAdapter{
+					Client:          o.RBACClient,
+					NamespaceClient: o.CoreClient.Namespaces(),
+				},
+			}
+			result, err := reconcileOptions.Run()
+			if err != nil {
+				return err
+			}
+			shallowInfoCopy.Object = result.RoleBinding.GetObject()
+			o.Print(&shallowInfoCopy)
+			return nil
+
+		case *rbac.ClusterRoleBinding:
+			reconcileOptions := reconciliation.ReconcileRoleBindingOptions{
+				Confirm:             true,
+				RemoveExtraSubjects: false,
+				RoleBinding:         reconciliation.ClusterRoleBindingAdapter{ClusterRoleBinding: t},
+				Client: reconciliation.ClusterRoleBindingClientAdapter{
+					Client: o.RBACClient.ClusterRoleBindings(),
+				},
+			}
+			result, err := reconcileOptions.Run()
+			if err != nil {
+				return err
+			}
+			shallowInfoCopy.Object = result.RoleBinding.GetObject()
+			o.Print(&shallowInfoCopy)
+			return nil
+
+		default:
+			// skip ignored resources
```

Comment:
nit, worth to `glog.V(n)` print the ignored ones?

Reply:
> nit, worth to glog.V(n) print the ignored ones?

done